### ISSUE TITLE
Publish test module for apm-agent-core

### ISSUE THREAD

**Issue by felixbarny** [elastic/apm-agent-java#1340]
Prerequisite for #937

**Comment by felixbarny** [comment]
We also need to publish apm-api-plugin so that the API can be used in tests

**Comment by jackshirazi** [comment]
I think we no longer need this

**Comment by jackshirazi** [comment]
agreed no longer needed as the community plugin regression testing can be done very well with the full agent embedded
